spaceship/lib/spaceship/connect_api/models/app.rb

```diff
@@ -203,7 +203,7 @@ def reject_version_if_possible!(client: nil, platform: nil)
       # @return (Bool) Was something changed?
       def ensure_version!(version_string, platform: nil, client: nil)
         client ||= Spaceship::ConnectAPI
-        app_store_version = get_edit_app_store_version(platform: platform)
+        app_store_version = get_edit_app_store_version(client: client, platform: platform)
```

Comment:
@BoaNeo Do you happen to have a stacktrace of the nil exception? We _shouldn’t_ need to pass in `client` here because ‘ get_edit_app_store_version’ will set `client` if its not passed in the same way this method does 🤔

Reply:
I'm no Ruby expert, I don't even like Ruby :), but unless there's some language-level magic going on that I don't understand, I think you are reading the code wrong.

If the client is not passed, you are correct that it will default to the same value in both functions, but that is not the use case that is broken. 

The case that doesn't work is when I do pass a client. In this case the code will use that client only in this function but then pass nil to get_edit_app_store_version (which will then use the default, which is probably nil if I'm passing in a client of my own, and in any case probably isn't the one I want it to use).

Does that make sense?

Reply:
OMG I DID READ IT WRONG 😛  I most likely didn’t have enough coffee in me when I reviewed this 🤦‍♂️ 

You right you right, this is 💯 Thanks for explaining this to me!